### PR TITLE
OnlineDDL: "cancel-all" command to cancel all pending migrations in keyspace

### PR DESCRIPTION
**Pull request by shlomi-noach**
This PR adds a `vtctl OnlineDDL <keyspace> cancel-all` command.

The command is pretty strong: it will cancel any pending migration for the given keyspace. A migration is `pending` if it's running or about to run, ie in one of these states:

- `queued`
- `ready`
- `running`

I have a specific use case for `cancel-all`, where there's need to re-evaluate the migrations for a given keyspace; I can think of another use case: some BigOperation needs to take place on the keyspace and the operations team wishes to abort pending migrations.

General context: #6926

### CODE REVIEW

**Comment by shlomi-noach** [comment]
Documentation PR: https://github.com/vitessio/website/pull/622

**Comment by harshit-gangal** [comment]
I think we might also want to cancel any ongoing migration as well. If it is realised that it cannot run at the moment may be because of wrong query or time of change is wrong or it can be grouped better with newer change, etc.

I suggest to add a parameter to this command which can be force cancel.

**Comment by shlomi-noach** [comment]
> I think we might also want to cancel any ongoing migration as well. 

This already exists: https://vitess.io/docs/user-guides/schema-changes/managed-online-schema-changes/#cancelling-a-migration

e.g. `vtctlclient OnlineDDL commerce cancel 2201058f_f266_11ea_bab4_0242c0a8b007`

will cancel a speciic running migration, or, if it's not running yet, will ensure it doesn't run.

`cancel-all`, suggested in this PR, is based on the above.

**Comment by harshit-gangal** [comment]
May be I miss read the description.

**Comment by shlomi-noach** [comment]
So, we already have a per-migration `cancel`; what this PR adds is "cancel any migration that is pending on this keyspace", which is a pretty massive and risky operation, but fits really well with an internal POC I'm running, and can be useful fo remergency situations in production.

**Review by harshit-gangal — APPROVED** [review]

File: go/test/endtoend/onlineddl/onlineddl_test.go [hunk 1]
```diff
@@ -182,6 +183,26 @@ func TestSchemaChange(t *testing.T) {
 		checkRecentMigrations(t, uuid, schema.OnlineDDLStatusFailed)
 		checkCancelMigration(t, uuid, false)
 		checkRetryMigration(t, uuid, true)
+		// migration will fail again
+	}
+	{
+		// no migrations pending at this time
+		time.Sleep(10 * time.Second)
+		checkCancelAllMigrations(t, 0)
```
[the remark this answers is not in the document]
Comment (shlomi-noach) [reply go/test/endtoend/onlineddl/onlineddl_test.go:191]: Unfortunately all the online-DDL flow works asynchronously, with check intervals. In this `endtoend` tests intervals are reduced to just a few secs, and still you have to give time for the check, for `gh-ost`, for callback...

This is why I extracted online-ddl to be on its own CI `shard`; it's still shorter than our other `endtoend` tests so right now I feel like it's not worth investing the optimization effort.

File: go/vt/vttablet/onlineddl/executor.go [hunk 2]
```diff
@@ -894,6 +895,32 @@ func (e *Executor) cancelMigrations(ctx context.Context, uuids []string) (err er
...
 }
 
+// cancelPendingMigrations cancels all pending migrations (that are expected to run or are running)
+// for this keyspace
+func (e *Executor) cancelPendingMigrations(ctx context.Context) (result *sqltypes.Result, err error) {
+	parsed := sqlparser.BuildParsedQuery(sqlSelectPendingMigrations, "_vt")
+	r, err := e.execQuery(ctx, parsed.Query)
+	if err != nil {
+		return result, err
+	}
```
[the remark this answers is not in the document]
Comment (shlomi-noach) [reply go/vt/vttablet/onlineddl/executor.go:905]: You're correct, and this race condition is unavoidable. If the user chooses to "cancel" or "cancel-all" just when the migration is about to complete, it's possible that the migration will win the race and complete; in which case the "cancel" is a no-op and nothing happens (the migration is just complete)

**Pull request merged**